sympy/polys/polytools.py

```diff
@@ -7938,3 +7937,258 @@ def named_poly(n, f, K, name, x, polys):
     else:
         poly = Poly.new(poly, head)
     return poly if polys else poly.as_expr()
+
+
+def factor_system_poly(polys, **kwargs):
+    """
+        Factor a system of polynomials and find their minimal factor combinations.
+
+        This function takes a list of polynomials and returns all minimal combinations
+        of factors that, when set to zero, satisfy the system of equations.
```

Comment:
The key thing to mention here is that the set of solutions to the original equations is the union of the sets of solutions to the factored systems.